models/orogen/skid4_control.rb

```diff
@@ -1,17 +1,19 @@
+require 'models/services/motion2d_control_loop'
+require 'models/services/joints_control_loop'
 require 'models/blueprints/control'
 
 class OroGen::Skid4Control::SimpleController
-    provides Base::ActuatorControllerSrv, :as => 'controller'
-    provides Base::Motion2DControlledSystemSrv, :as => 'controlled_system'
+    provides Rock::Services::JointsOpenLoopController, :as => 'controller'
```

Comment:
About the 'as:' syntax. You know most of it, but let's describe it for everyone.

Ruby pre-2.0 did not have a keyword argument syntax. It was emulated with a trick of the parser that if an argument list ended with something that looks like a hash, it would be "gathered" within a hash and passed to the method, e.g. a method defined as

~~~
method(options = Hash.new)
~~~

and a call looking like

~~~
method(:arg => 'a', :remove => 'b')
~~~

was converted by the parser into

~~~
method(Hash[:arg => 'a', :remove => 'b'])
~~~

2.0 introduced keyword arguments. The keyword arguments are specified on the method interface with `arg: 'a'`

~~~
def method(arg: nil, remove: false)
~~~

and called with

~~~
method(arg: 'a', remove: false)
~~~

to ease the transition from option hashes to keyword argument syntax (which are very widely used), they allow to use the hash syntax to call a method that uses keyword arguments. I don't know how long they're going to allow that (it's pretty complex parser and runtime-wise). Moreover, CRuby recently had optimization work for calling methods with keyword arguments, and I don't know if it applies only to the new syntax or to any type of calls.

Plus, I like the new syntax better. So, better to replace the old syntax by the new one whenever we write new code / adapt old code.